public/docs/ts/latest/guide/reactive-forms.jade

```diff
@@ -0,0 +1,885 @@
+  Piping it through the `JsonPipe` renders the model as JSON in the browser:
+
+figure.image-display
+  img(src="/resources/images/devguide/reactive-forms/json-output.png" width="400px" alt="JSON output")
+
+:marked
+  The initial `name` property value is the empty string. 
+  Type into the _name_ input box and watch the keystokes appear in the JSON.
+
+
+
+:marked
+  Great! You have the basics of a form. 
+  
+  In real life apps, forms get big fast. 
+  The way you are creating `FormControls` currently is both tedious and hard to maintain. 
+  `FormBuilder` makes form development and maintenance easier.
+
+
+.l-main-section
+a#formbuilder
+:marked
+  ## Introduction to _FormBuilder_
+
+  The `FormBuilder` class helps reduce repetition and 
+  clutter by handling details of control creation for you. 
+
+  You already imported `FormBuilder` into `hero-detail.component.ts`.
+  Use it now to refactor the `HeroDetailComponent` into something that's a little easier to read and write,
+  by following this plan:
+
+  * Explicitly declare the type of the `heroForm` property to be `FormGroup`; you'll initialize it later.
+  * Inject a `FormBuilder` into the constructor.
+  * Add a new `createForm` method that uses the `FormBuilder` to define the `heroForm`.
+  * Call `createForm` in the constructor.
+
+  The revised `HeroDetailComponent` looks like this:
++makeExample('reactive-forms/ts/app/hero-detail-versions.component.ts', 'v2','app/hero-detail.component.ts (excerpt)')(format=".")
+
+:marked
+  `FormBuilder.group` is a factory method that creates a `FormGroup`. &nbsp;
+  `FormBuilder.group` takes an object whose keys and values are `FormControl` names and their definitions. 
+  In this example, the `name` control is defined by its initial data value, an empty string.
+
+  Defining a group of controls in a single object makes for a compact, readable style.
+  It beats writing an equivalent series of `new FormControl(...)` statements.
+
+:marked
+  ### More FormControls
+  A hero has more than a name. A hero has an address too. 
+  Add some address `FormControls` to the `heroForm` as follows.
+
++makeExample('reactive-forms/ts/app/hero-detail-versions.component.ts', 'v3','app/hero-detail.component.ts (excerpt)')(format=".")
+
+:marked
+  Then add corresponding markup in `hero-detail.component.html` 
+  within the `form` element.
+
++makeExample('reactive-forms/ts/app/hero-detail-3.component.html', '','app/hero-detail.component.html')(format=".")
+
+.alert.is-helpful
+  :marked
+    *Reminder*: Ignore the many mentions of `form-group` and `form-control`in this markup.
+    Those are _bootstrap_ CSS classes that Angular itself ignores.
+    Pay attention to the `formGroupName` and `formControlName` attributes.
+    They are the Angular directives that bind the HTML controls to the 
+    Angular `FormGroup` and `FormControl` properties in the component class.
+
+.l-main-section
+a#grouping
+:marked
+  ### Group FormControls
```

Comment:
How about 'Nested FormGroups' ? When I first scanned this, I read the title as a noun and thought you were phrasing "FormGroup" in a different way.